samples: matter: Enabled use of OT without Zephyr L2 (#22952)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Matter
------

* Added:

  * FastTrack Recertification and Portfolio Certification programs.
  * Support for configuring the Matter over Thread applications to use OpenThread API directly instead of utilizing Zephyr L2 networking layer.

>     This allows for a significant memory usage reduction in Matter applications - approximately 15 kB of RAM and 40 kB of flash.

Author: Hmm, isn't it actually the opposite? The numbers should be fairly constant, as we just disabled some abstract portion of software. The percentage depends on the device, as available space is different depending on the SoC.

Matter
------

* Added:

>   * FastTrack Recertification and Portfolio Certification programs.

Reviewer: ```suggestion
  * FastTrack recertification and portfolio certification programs.
```

>   * Support for configuring the Matter over Thread applications to use OpenThread API directly instead of utilizing Zephyr L2 networking layer.

Reviewer: ```suggestion
  * Matter-over-Thread apps can now use the OpenThread API directly, bypassing the Zephyr L2 layer.
```

Author: Applied, but replaced bypassing with `instead of using intermediate Zephyr L2 layer.`, because we are not bypassing it, we just disabled it at all.

>     This allows for a significant memory usage reduction in Matter applications.

Reviewer: ```suggestion
    This change reduces memory usage in Matter applications.
```

>     For example, with the :zephyr:board:`nrf54l15dk` it allowed to save approximately 15 kB of RAM and 40 kB of flash.

Reviewer: ```suggestion
    On the :zephyr:board:`nrf54l15dk`, it saves approximately 15 kB of RAM and 40 kB of flash.
```
